ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -269,7 +294,13 @@ protected BlockCreationResult createBlock(
           new RequestProcessingContext(blockProcessingContext, transactionResults.getReceipts());
 
       Optional<List<Request>> maybeRequests =
-          requestProcessor.map(processor -> processor.process(requestProcessingContext));
+          requestProcessor.map(
+              processor -> processor.process(requestProcessingContext, postExecutionAccessList));
+
+      if (postExecutionUpdater instanceof StackedUpdater<?, ?> stackedUpdater) {
```

Comment:
instead of having this check everywhere 

> if (postExecutionUpdater instanceof StackedUpdater<?, ?> stackedUpdater) {

why not having this check during the BAL initialization and disable it for the JournaledUpdater mode